nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-shared/src/main/java/org/apache/nifi/kafka/shared/login/IAMLoginConfigProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.kafka.shared.login;
+
+import org.apache.nifi.context.PropertyContext;
+import org.apache.nifi.kafka.shared.component.KafkaClientComponent;
+import org.apache.nifi.util.StringUtils;
+
+/**
+ * SASL AWS IAM Login Module implementation of configuration provider
+ */
+public class IAMLoginConfigProvider implements LoginConfigProvider {
```

Comment:
Recommend adjusting the class name to be more specific to AWS MSK:

```suggestion
public class AwsMskIAMLoginConfigProvider implements LoginConfigProvider {
```

Reply:
Just a minor thing, I changed `IAM` to `Iam`, because normally AWS and MSK should be written with capitals, so if I use camelcase there, then I think I should use that for IAM too.

Reply:
Thanks, that sounds good, `AwsMskIamLoginConfigProvider` is a bit easier to read than `AWSMSKIAMLoginConfigProvider`.